### ISSUE TITLE
Literal identifiers cause macro errors when referenced inside @react components

### ISSUE THREAD

**Issue by hejfelix** [shadaj/slinky#117]
```scala
@js.native
trait Color extends js.Object {
  val `50`: String                 = js.native
  val `100`: String                = js.native
  val `200`: String                = js.native
  val `300`: String                = js.native
  val `400`: String                = js.native
...
}

@react class App extends StatelessComponent {
  type Props = Unit

  private val css = AppCSS

  println(MaterialUi.colors.blue.`300`)
...
}
```

```scala
[info] Compiling 1 Scala source to /home/fhargreaves/repos/slinky-wrappers/demo/target/scala-2.12/classes ...
[error] <macro>:8:35: ')' expected but double literal found.
[error]     println(MaterialUi.colors.blue.300)
[error]                                   ^
[error] <macro>:9:92: ';' expected but ')' found.
[error]     private val themeSettings = literal(palette = literal(primary = MaterialUi.colors.blue))
[error]                                                                                            ^
[error] two errors found
[error] (Compile / compileIncremental) Compilation failed
[error] Total time: 1 s, completed Mar 4, 2018 8:31:33 PM
```

**Comment by shadaj** [comment]
@hejfelix this looks like a bug in Scalameta. I'll try to see if this can be fixed, but in the mean time, you can use names like `_50` that don't depend on using literal identifiers.

**Comment by hejfelix** [comment]
Will that work? It's a part of a facade to a js object with those exact id's.

**Comment by shadaj** [comment]
Oh, just noticed that. You should be able to use `@JSName` to specify the id in the original object.

**Comment by hejfelix** [comment]
Cool. Didn't know about that :) thanks so far

**Comment by shadaj** [comment]
Just tested, this is fixed in `0.5.0`!
